### PR TITLE
chore(ci): Add perms for PR labeler

### PR DESCRIPTION
**Pull request by bsmth**
Action is 403 at the moment, adding perms to the workflow:

- [x] https://github.com/mdn/content/pull/31729
- Logs https://github.com/mdn/content/actions/runs/7541990630/job/20529845639?pr=31684

### CODE REVIEW

**Comment by OnkarRuikar** [comment]
The trigger was `pull_request_target` before. In case of `pull_request` the `GITHUB_TOKEN` doesn't get write perms. You need to either change the trigger or provide token with elevated permissions.

**Comment by bsmth** [comment]
Thanks @OnkarRuikar 

> The trigger was `pull_request_target` before.

That's right, I was hoping to get rid of this.

>provide token with elevated permissions

I'm assuming the changes in this PR cover that, right?

**Comment by bsmth** [comment]
I guess I have to revert to `pull_request_target`: https://github.com/actions/labeler#permissions

**Comment by OnkarRuikar** [comment]
> I'm assuming the changes in this PR cover that, right?

Nope, `permissions:` part is further tightening on already available permissions to the `GITHUB_TOKEN`.

File: .github/workflows/pr-labeler.yml [hunk 1]
```diff
@@ -1,10 +1,7 @@
 name: Pull request labeler
 
 on:
-  pull_request:
-    types: [opened, synchronize]
-    branches:
-      - "main"
+  - pull_request_target
```
Comment (LeoMcA) [inline .github/workflows/pr-labeler.yml:4]: I think a hybrid approach from the commits works (I'm not sure we need the issues permission though):

```suggestion
  - pull_request_target
  
 permissions:
  pull-requests: write
  contents: read
```

An approach like this with `pull_request_target` is in the github docs for a similar situation to ours: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input.

And to lock things down even more, I think we want to add `persist-credentials: false` to the checkout step, to ensure the key doesn't get cached locally anywhere.

**Review by LeoMcA — CHANGES_REQUESTED** [review]

**Comment by bsmth** [comment]
Going to merge to get CI green as some other PRs are blocked by this now. Thank you, both!

**Pull request merged**